packages/fetchai/connections/p2p_libp2p_client/connection.py

```diff
@@ -251,7 +251,19 @@ async def _setup_connection(self) -> None:
         buf = msg.SerializeToString()
         await self._send(buf)
 
-        buf = await self._receive()
+        self.logger.debug("Waiting for registration message...")
+        try:
+            buf = await self._read_message_from_reader()
+        except ConnectionError as e:  # pragma: nocover
+            self.logger.error(f"Connection error: {e}.")
+            raise e
+        except IncompleteReadError as e:  # pragma: no cover
+            self.logger.error(
+                "Connection disconnected while reading from node ({}/{})".format(
+                    len(e.partial), e.expected
+                )
+            )
+            raise e
```

Comment:
Calling _receive here can lead to problems if connect call not finished yet..., need to let exception propagate